python_modules/dagster/dagster/_core/definitions/sensor_definition.py

```diff
@@ -366,17 +463,29 @@ def __call__(self, *args, **kwargs):
                 )
 
             context = context if context else build_sensor_context()
-
-            return self._raw_fn(context)
+            context_param = {context_param_name: context}
 
         else:
-            if len(args) + len(kwargs) > 0:
-                raise DagsterInvalidInvocationError(
-                    "Sensor decorated function has no arguments, but arguments were provided to "
-                    "invocation."
+            # We still take a context arg even if the underlying function doesn't require it
+            # this is so that we can pass resources if the sensor needs any
+            context: Optional[SensorEvaluationContext] = None
+            if args:
+                context = check.opt_inst_param(args[0], "context", SensorEvaluationContext)
+            elif kwargs:
+                context = check.opt_inst_param(
+                    list(kwargs.values())[0], "context", SensorEvaluationContext
                 )
+            context = context if context else build_sensor_context()
 
-            return self._raw_fn()
+        resources = _validate_and_get_resource_dict(
+            context, self.name, self._required_resource_keys
+        )
+        return self._raw_fn(**context_param, **resources)
```

Comment:
This threw me for a loop a little bit. This relies knowing about an assignment 40 lines up to see that it will actually complete execution and that the context_param dictionary will only include a value if the `if context_param_name:` predicate passes.

I think you should be able to collapse this into a smaller chunk of code (with a single codepath for context creation) that simulateneously feels less clever.